### PR TITLE
OpenAPI runner

### PR DESCRIPTION
**Pull request by Saarett**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Created an OpenAPI runner, 6 checks, unit tests and contribution documentation.
The runner is a subclass of @schosterbarak 's `YamlRunner` and `JsonRunner` classes
The registry "object_registry" fits json/yaml formats.

### CODE REVIEW

**Comment by schosterbarak** [comment]
@Saarett should the new class be derrived from the YamlRunner or JsonRunner?

**Comment by gruebel** [comment]
@Saarett can we close this?

**Comment by Saarett** [comment]
> @Saarett can we close this?

Not yet, we are about to merge some new checks 💪

File: tests/terraform/util/test_doc_generator.py [hunk 1]
```diff
@@ -20,9 +20,9 @@ def test_get_checks_returned_check_number():
     "input_frameworks,expected_frameworks",
     [
         (["all"], {"arm", "Bicep", "Cloudformation", "dockerfile", "Kubernetes", "secrets", "serverless", "Terraform",
-                   "github_configuration", "gitlab_configuration", "bitbucket_configuration", "github_actions"}),
+                   "github_configuration", "gitlab_configuration", "bitbucket_configuration", "github_actions", "OpenAPI"}),
         (None, {"arm", "Bicep", "Cloudformation", "dockerfile", "Kubernetes", "secrets", "serverless", "Terraform",
-                "github_configuration", "gitlab_configuration", "bitbucket_configuration", "github_actions"}),
+                "github_configuration", "gitlab_configuration", "bitbucket_configuration", "github_actions", "OpenAPI"}),
```
Comment (schosterbarak) [inline tests/terraform/util/test_doc_generator.py:25]: is it case sensitive?

Comment (schosterbarak) [reply tests/terraform/util/test_doc_generator.py:25]: so let's do it lower case

File: checkov/common/util/docs_generator.py [hunk 2]
```diff
@@ -95,6 +96,8 @@ def add_from_repository(registry: Union[BaseCheckRegistry, BaseGraphRegistry], c
     if any(x in framework_list for x in ("all", "bicep")):
         add_from_repository(bicep_param_registry, "parameter", "Bicep")
         add_from_repository(bicep_resource_registry, "resource", "Bicep")
+    if any(x in framework_list for x in ("all", "openapi")):
+        add_from_repository(openapi_registry, "resource", "OpenAPI")
```
Comment (schosterbarak) [inline checkov/common/util/docs_generator.py:100]: let's do lower case

File: tests/terraform/util/test_doc_generator.py (hunk 1 above)
Comment (ayajbara) [reply tests/terraform/util/test_doc_generator.py:25]: its for docs only

File: checkov/common/util/docs_generator.py (hunk 2 above)
Comment (ayajbara) [reply checkov/common/util/docs_generator.py:100]: its for docs only

**Review by nimrodkor — APPROVED** [review]

**Pull request merged**